mux/src/tmux.rs

```diff
@@ -139,12 +145,9 @@ impl TmuxDomainState {
                         let cmd = self.cmd_queue.borrow_mut().pop_front().unwrap();
                         let domain_id = self.domain_id;
                         *self.state.borrow_mut() = State::Idle;
-                        promise::spawn::spawn(async move {
-                            if let Err(err) = cmd.process_result(domain_id, &response) {
-                                log::error!("error processing result: {}", err);
-                            }
-                        })
-                        .detach();
+                        if let Err(err) = cmd.process_result(domain_id, &response, &mut self.inner.borrow_mut()) {
```

Comment:
I don't know enough Rust to make it compile. It asks for 'static' lifetime of `self`.

I see the intention here is to not block the current thread while processing the data. I feel this right now is also OK? Because it helps debugging a bit if it does NOT switch to another thread. Plus no one can really use tmux-cc at this moment.

Reply:
The CI didn't get to compile this because the changes weren't `cargo fmt`'d, so I can't see the actual error message :-(

I think you might need to *reborrow* this by adding a `*` to deref the `RefMut` to the underlying state struct: `&mut* self.inner.borrow_mut()`, but it may be the case that you need to put the `RefMut` into a temporary to keep it alive.

I'm just speculating though!

Reply:
Sry I still can't make it work :sweat: . It tells me:

```text
`self` has an anonymous lifetime `'_` but it needs to satisfy a `'static` lifetime requirement
```